src/functional/otf/compilation/build_data.py

```diff
@@ -0,0 +1,79 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
+
+import dataclasses
+import enum
+import json
+import pathlib
+from typing import Final, Optional
+
+
+_DATAFILE_NAME: Final = "gt4py.json"
+
+BuildStatus = enum.IntEnum("BuildStatus", ["STARTED", "CONFIGURED", "COMPILED"], start=0)
+
+
+@dataclasses.dataclass(frozen=True)
+class BuildData:
+    """
+    Tracks build status and results of GT4Py program compilations.
+
+    Build systems projects should keep this information up to date for each build,
+    stored to a json file in the build folder at all times.
+    """
+
+    status: BuildStatus
+    module: pathlib.Path
```

Comment:
```suggestion
    binary: pathlib.Path
```

Reply:
What is the rationale behind replacing the description "this is a python module and can be imported" with the less specific description "this is a file in binary format"?

Reply:
I would leave it and add an `other_binaries` attribute to `BuildData` when there is a usecase. Currently the `.module` attribute pointing to an existing file is used to mean it can be imported and must contain a callable Python object with name `.entry_point_name`.

While in principle the build systems are general enough, supporting code would be required to make it useful. Such work would then include extending `BuildData` if appropriate.

Alternatively the current semantics could be changed but the value of that is questionable, since there is no usecase.

Reply:
I clarified this in the `BuildData` docstring.